okhttp/src/main/java/okhttp3/UnmodifiableList.java

```diff
@@ -0,0 +1,46 @@
+package okhttp3;
+
+import java.util.Iterator;
+import java.util.List;
+
+public final class UnmodifiableList<T> implements Iterable<T> {
```

Comment:
Not a fan of new public class for the project, but hadn't found a better solution yet. But, looks like it can be used not only for interceptors but also for other things too both internally and for the public API.